crates/core/src/checkpoint.rs

```diff
@@ -57,22 +57,25 @@ FROM bucket_list
      bucket_list.bucket = oplog.bucket AND
      oplog.op_id <= CAST(json_extract(?1, '$.last_op_id') as INTEGER) AND
      oplog.op_id > bucket_list.lower_op_id
-GROUP BY bucket_list.bucket")?;
+GROUP BY bucket_list.bucket",
+    )?;
 
     statement.bind_text(1, data, sqlite::Destructor::STATIC)?;
 
     let mut failures: Vec<String> = alloc::vec![];
 
     while statement.step()? == ResultCode::ROW {
         let name = statement.column_text(0)?;
+        // checksums with column_int are wrapped to i32 by SQLite
         let add_checksum = statement.column_int(1)?;
         let oplog_checksum = statement.column_int(2)?;
         let _count = statement.column_int(3)?;
         let _last_op_id = statement.column_int64(4)?;
         let _last_applied_op = statement.column_int64(5)?;
         let expected_checksum = statement.column_int(6)?;
 
-        let checksum = oplog_checksum + add_checksum;
+        // wrapping add is like +, but safely overflows
+        let checksum = oplog_checksum.wrapping_add(add_checksum);
```

Comment:
Rust default overflow semantics for i32 is panic in debug mode, and do a wrapping add in release mode. This is why it previously worked in release mode, but could crash hard in debug mode.

`wrapping_add` fixes this to work correctly in both debug and release mode.